src/modules/txbuilder/RadixTransactionBuilder.ts

```diff
@@ -468,93 +513,97 @@ export default class RadixTransactionBuilder {
         return this
     }
 
+
     /**
      * Builds the atom, finds a node to submit to, adds network fee, signs the atom and submits
      * @param signer
      * @returns a BehaviourSubject that streams the atom status updates
      */
     public signAndSubmit(signer: RadixSignatureProvider) {
+        const atom = this.buildAtom()
+        
+        const stateSubject = new BehaviorSubject<string>('FINDING_NODE')
+
+        // Find a shard, any of the participant shards is ok
+        const shard = atom.getAddresses()[0].getShard()
+        
+        // Get node from universe
+        radixUniverse.getNodeConnection(shard)
+            .then(connection => {
+                RadixTransactionBuilder.signAndSubmitAtom(atom, connection, signer, this.participants.values())
+                    .subscribe(stateSubject)
+            })
+
+        return stateSubject
+    }
+
+    public buildAtom() {
         if (this.particleGroups.length === 0) {
-            throw new Error('No particles specified')
+            throw new Error('No particle groups specified')
         }
 
         const atom = new RadixAtom()
-
         atom.particleGroups = this.particleGroups
 
         // Add timestamp
         atom.setTimestamp(Date.now())
+        return atom
+    }
 
-        // Find a shard, any of the participant shards is ok
-        const shard = atom.getAddresses()[0].getShard()
-
-        // Get node from universe
-        let nodeConnection: RadixNodeConnection = null
-
-        const stateSubject = new BehaviorSubject<string>('FINDING_NODE')
-
+    public static signAndSubmitAtom(atom: RadixAtom, connection: RadixNodeConnection, signer: RadixSignatureProvider, participants: RadixAccount[]) {
         let signedAtom = null
+        
+        // Add POW fee
+        const endorsee = RadixAddress.fromPublic(connection.node.nodeInfo.system.key.bytes)
+
+        const stateSubject = new BehaviorSubject<string>('GENERATING_POW')
+        
+        RadixFeeProvider.generatePOWFee(
+            radixUniverse.universeConfig.getMagic(),
+            radixUniverse.powToken,
+            atom,
+            endorsee,
+        ).then(powFeeParticle => {
+            const powFeeParticleGroup = new RadixParticleGroup([RadixSpunParticle.up(powFeeParticle)])
+            atom.particleGroups.push(powFeeParticleGroup)
+
+            // Sign atom
+            stateSubject.next('SIGNING')
+            return signer.signAtom(atom)
+        }).then(_signedAtom => {
+            signedAtom = _signedAtom
+
+            logger.debug(signedAtom.hid.toString())
```

Comment:
Is this left here by intention? The reason I'm wondering is that there is no further information in the log regarding what this is. Just a naked, raw hash.

Reply:
It's sometimes useful to me, but you're right it would be completely useless for anyone else.